python/cuml/linear_model/linear_regression.pyx

```diff
@@ -303,8 +326,15 @@ class LinearRegression(LinearPredictMixin,
 
         """
         cdef uintptr_t X_ptr, y_ptr, sample_weight_ptr
+
+        need_explicit_copy = self.copy_X and hasattr(X, "__cuda_array_interface__") \
+            and (len(X.shape) < 2 or X.shape[1] == 1)
```

Comment:
Does the issue only appear under these conditions?

Reply:
I think I responded to a previous version of your question. Yes, an implicit copy is always created when the input is not a vector.

Reply:
Would it not make more sense to have this logic be part of the `input_to_cuml_array` utility function?

Reply:
In principle yes, but that would potentially change the behavior of many more classes and would require more thorough investigation and testing.